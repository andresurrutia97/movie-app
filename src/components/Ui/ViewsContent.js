import React from 'react';
import PropTypes from 'prop-types';

const ViewsContent = ({ children }) => {
  return (
    <div className="pt-16 my-4 mx-5 md:w-6/12 md:mx-auto md:mt-16">
      {children}
    </div>
  );
};

ViewsContent.propTypes = {
  children: PropTypes.node,
};

ViewsContent.defaultProps = {
  children: null,
};

export default ViewsContent;
