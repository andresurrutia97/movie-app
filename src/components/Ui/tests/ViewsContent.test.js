import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';

import ViewsContent from '../ViewsContent';

describe('<ViewsContent/>', () => {
  const component = (
    <ViewsContent>
      <div>Test</div>
    </ViewsContent>
  );

  it('should render correctly ', () => {
    const tree = renderer.create(component).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should coantain a <div/> as a children', () => {
    const wrapper = shallow(component);
    expect(wrapper.children().find('div')).toHaveLength(1);
    expect(wrapper.children().find('div').props().children).toBe('Test');
  });
});
