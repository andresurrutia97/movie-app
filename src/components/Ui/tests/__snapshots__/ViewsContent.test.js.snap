// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ViewsContent/> should render correctly  1`] = `
<div
  className="pt-16 my-4 mx-5 md:w-6/12 md:mx-auto md:mt-16"
>
  <div>
    Test
  </div>
</div>
`;
